lib/ansible/modules/network/enos/enos_facts.py

```diff
@@ -0,0 +1,497 @@
+    type: dict
+  ansible_net_neighbors:
+    description: The list of LLDP neighbors from the remote device
+    returned: when interfaces is configured
+    type: dict
+'''
+
+import re
+
+from ansible.module_utils.enos import run_commands, enos_argument_spec, check_args
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.six.moves import zip
+
+
+class FactsBase(object):
+
+    COMMANDS = list()
+
+    def __init__(self, module):
+        self.module = module
+        self.facts = dict()
+        self.responses = None
+        self.PERSISTENT_COMMAND_TIMEOUT = 60
+
+    def populate(self):
+        self.responses = run_commands(self.module, self.COMMANDS,
+                                      check_rc=False)
+
+    def run(self, cmd):
+        return run_commands(self.module, cmd, check_rc=False)
+
+
+class Default(FactsBase):
+
+    COMMANDS = ['show version', 'show run']
+
+    def populate(self):
+        super(Default, self).populate()
+        data = self.responses[0]
+        data_run = self.responses[1]
+        if data:
+            self.facts['version'] = self.parse_version(data)
+            self.facts['serialnum'] = self.parse_serialnum(data)
+            self.facts['model'] = self.parse_model(data)
+            self.facts['image'] = self.parse_image(data)
+        if data_run:
+            self.facts['hostname'] = self.parse_hostname(data_run)
+
+    def parse_version(self, data):
+        match = re.search(r'^Software Version (.*?) ', data, re.M | re.I)
+        if match:
+            return match.group(1)
+
+    def parse_hostname(self, data_run):
+        for line in data_run.split('\n'):
+            line = line.strip()
+            match = re.match(r'hostname (.*?)', line, re.M | re.I)
+            if match:
+                hosts = line.split()
+                hostname = hosts[1].strip('\"')
+                return hostname
+            else:
+                return "NA"
+
+    def parse_model(self, data):
+        # match = re.search(r'^Cisco (.+) \(revision', data, re.M)
+        match = re.search(r'^Lenovo RackSwitch (\S+)', data, re.M | re.I)
+        if match:
+            return match.group(1)
+
+    def parse_image(self, data):
+        match = re.search(r'(.*) image1(.*)', data, re.M | re.I)
+        if match:
+            return "Image1"
+        else:
+            return "Image2"
+
+    def parse_serialnum(self, data):
+        # match = re.search(r'board ID (\S+)', data)
+        match = re.search(r'^Switch Serial No:  (\S+)', data, re.M | re.I)
+        if match:
+            return match.group(1)
+
+
+class Hardware(FactsBase):
+
+    COMMANDS = [
+        'show system memory'
+    ]
+
+    def populate(self):
+        super(Hardware, self).populate()
+        data = self.responses[0]
+        # file = open('Sunil.txt', "a")
+        # file.write(data)
+        # file.close()
```

Comment:
This seems to debug code. Please remove the commented lines.

Reply:
Done